src/async_impl/response.rs

```diff
@@ -194,6 +187,19 @@ impl Response {
         }
     }
 
+    /// Return the encoding declared in the `Content-Type` header, if any.
+    ///
+    /// This can be used with a library such as [`encoding_rs`] to perform
+    /// manual decoding if `text` and `text_with_charset` are too restrictive.
+    ///
+    /// [`encoding_rs`]: https://docs.rs/encoding_rs
+    pub fn charset(&self) -> Option<String> {
```

Comment:
I'd probably wish to return `&str`, but I know why you didn't (can't borrow from the Mime)...

Reply:
The best I could think of was an additional private function that returns the `Mime`, to avoid the clone for `text_with_charset`, but I figured it wasn't worth it to save the allocation.